modules/frontend/slos.go

```diff
@@ -0,0 +1,119 @@
+package frontend
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+const (
+	traceByIDOp = "traces"
+	searchOp    = "search"
```

Comment:
Do we want to track SLOs on the other frontend operation?  Looks like we lost a couple.

Reply:
We currently don't support "policy based" SLOs for those other ops and just use `tempo_request_duration_seconds` histograms. Only these two operations were using these metrics for SLOs so I removed the other two to align `tempo_query_frontend_queries_total` and `tempo_query_frontend_queries_within_slo_total`. Previously, `tempo_query_frontend_queries_total` had more values for the `op` label then `tempo_query_frontend_queries_within_slo_total` and I wanted this pair of metrics to be aligned.